src/repr/src/scalar.rs

```diff
@@ -2757,6 +2830,120 @@ fn arb_list(element_strategy: BoxedStrategy<PropDatum>) -> BoxedStrategy<PropLis
         .boxed()
 }
 
+#[derive(Debug, PartialEq, Clone)]
+pub struct PropRange(
+    Row,
+    Option<(
+        (Option<Box<PropDatum>>, bool),
+        (Option<Box<PropDatum>>, bool),
+    )>,
+);
+
+fn arb_range_data() -> BoxedStrategy<(PropDatum, PropDatum)> {
+    prop_oneof![(
+        any::<i32>().prop_map(PropDatum::Int32),
+        any::<i32>().prop_map(PropDatum::Int32),
+    ),]
+    .boxed()
+}
+
+#[allow(dead_code)]
+fn arb_range() -> BoxedStrategy<PropRange> {
```

Comment:
I would expect to see this in the 
```rust
let leaf = prop_oneof![
    // ...
]
```
section of the `arb_datum` implementation.

Reply:
ty!